cfme/tests/cli/test_appliance_console.py

```diff
@@ -773,27 +773,28 @@ def test_appliance_console_restart():
         testSteps:
             1. 'ap' launches appliance_console.
             2. RETURN clears info screen.
-            3. '16' Stop EVM Server Processes.
-            4. 'Y' Stop CFME Server Gracefully.
-            5. Wait for some time.
-            6. '17' Start EVM Server Processes.
-            7. 'Y' to Start CFME server.
-            8. Wait for some time.
-            9. '21' Quit from appliance_console menu.
-            10. Open appliance IP into Browser.
+            3. '18' restart appliance.
+            4. '1' Restart.
+            5. Verify UI (Open appliance IP into Browser.)
         expectedResults:
             1.
             2.
             3.
             4.
-            5. Cross-check service stopped.
-            6.
-            7.
-            8. Cross-check service started.
-            9.
-            10. Confirm that Appliance is running into browser.
+            5. Confirm that Appliance is running into browser.
     """
-    pass
+    appliance = temp_appliance_preconfig_funcscope
+    restart_appliance_cmd = ("ap", RETURN, app_con_menu["restart_app"], "1", TimedCommand("Y", 0))
+    appliance.appliance_console.run_commands(restart_appliance_cmd, timeout=30)
+
+    wait_for(
```

Comment:
Just as a note:

There is a `mgmt` property on the Appliance instance, that gives you a wrapanapi instance directly from the provider hosting the appliance itself. This can be used to detect the power state of the appliance - but its not available on IPAppliance instances, which are not connected to the provider hosting the appliance.

Its a better way to detect whether or not the reboot occurred, but its limited in scope that it supports. Because this is using the temp_appliance fixture, you can be fairly certain that the appliance instance is an Appliance class, but if someone were to create a temp_appliance backing method that used regular IPAppliance, the test case wouldn't work.

The test could look for the `mgmt` property, and use the actual power state if possible, falling back to this net_check method if necessary.